gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2

```diff
@@ -918,7 +954,13 @@ def test_{{ method_name }}_pager(transport_name: str = "grpc"):
         {% endif %}
         pager = client.{{ method_name }}(request={})
 
-        assert pager._metadata == metadata
+        {% if service.version %}
+        if HAS_GOOGLE_API_CORE_VERSION_HEADER:
+            expected_metadata = tuple(expected_metadata) + (
+                version_header.to_api_version_header("{{ service.version }}"),
+            )
+        {% endif %}
+        assert pager._metadata == expected_metadata
```

Comment:
It seems this would be true by accident, in the following sense: Tuples are equal if they're equal element-wise; order matters. However, HTTP headers are not guaranteed to be in a particular order. So here the test would pass because the order in which you set the expectations happens happens to match the order in which the underlying library put the headers in the HTTP requests. If anything changes (the order in which the headers are set, or the order in which we add expectations), this would break. So we should be more robust about this as well. Let's include this in the issue about structuring headers checks.

Thoughts?

Reply:
Filed https://github.com/googleapis/gapic-generator-python/issues/2024